client/routes/workflow/components/cytoscape-graph.vue

```diff
@@ -0,0 +1,237 @@
+import { getEventConnections } from "../helpers/get-event-connections";
+import graphStyles from "../helpers/graph-styles";
+import store from "../../../store/index";
+import cytoscape from "cytoscape";
+import omit from "lodash-es/omit";
+
+cytoscape.use(dagre);
+
+export default {
+  name: "cytoscape-graph",
+  props: ["events"],
+  data() {
+    return {
+      nodes: [],
+      edges: [],
+      styles: graphStyles,
+      parentArray: []
+    };
+  },
+  watch: {
+    selectedEvent(id) {
+      if (id) this.selectNode(id);
+    }
+  },
+  methods: {
+    zoomToNode(node) {
+      let zoom = 1.1,
+        bb = node.boundingBox(),
+        w = cy.width(),
+        h = cy.height(),
+        pan = {
+          x: (w - zoom * (bb.x1 + bb.x2)) / 2,
+          y: (h - zoom * (bb.y1 + bb.y2)) / 2
+        };
+
+      //Pan the graph to view node
+      cy.animate({
+        zoom: 1.1,
+        pan: pan
+      });
+    },
+    updateChildBtn(node) {
+      let nodeData = node.data();
+      if (nodeData.childRoute) {
+        store.commit("childRoute", {
+          route: nodeData.childRoute,
+          btnText: "To child"
+        });
+      } else if (nodeData.newExecutionRunId) {
+        store.commit("childRoute", {
+          route: nodeData.newExecutionRunId,
+          btnText: "Next execution"
+        });
+      } else {
+        store.commit("toggleChildBtn");
+      }
+    },
+    selectNode(id) {
+      //Deselect all previously selected nodes
+      cy.$(":selected").deselect();
+
+      //Mark current node as selected
+      let node = cy.elements("node#" + id);
+      node.select();
+
+      this.updateChildBtn(node);
+      this.zoomToNode(node);
+    },
+    async buildTree() {
+      this.events.forEach(event => {
+        let {
+          parentWorkflowExecution,
+          previousExecutionRunId,
+          newExecutionRunId,
+          status,
+          childRoute
+        } = getEventConnections(event, this.events);
+
+        //We are viewing a child workflow, show parent btn
+        if (previousExecutionRunId) {
+          store.commit("previousExecutionRoute", previousExecutionRunId);
+        } else if (parentWorkflowExecution) {
+          store.commit("parentRoute", parentWorkflowExecution);
+        }
+
+        this.nodes.push({
+          data: {
+            id: event.eventId,
+            name: event.eventType,
+            childRoute: childRoute,
+            newExecutionRunId: newExecutionRunId,
+            status: status
+          }
+        });
+      });
+      //Set the direct and inferred relationships
+      this.events.forEach(node => {
+        this.setDirectAndInferred(node);
+      });
+
+      //Set the chronological relationships.
+      //If the node is not referred to as a parent it should be connected back to the graph with a chron child
+      this.events.forEach(node => {
+        if (!this.parentArray.includes(node.eventId)) {
+          this.setChron(node);
+        }
+      });
```

Comment:
again all above code can be moved out of the component. essentially you want to get to a point where the component just grabs all information it needs to render from the store via getters. Your component will get to a point which is purely render + onClick callbacks which just call mapMutation or connect functions.